core/src/replay_stage.rs

```diff
@@ -2436,6 +2457,7 @@ impl ReplayStage {
 
     fn mark_slots_confirmed(
         confirmed_forks: &[Slot],
```

Comment:
Could this be slot hashes instead to avoid the extra bank forks locking and assertions below?

Reply:
yup, much cleaner, updated here: https://github.com/solana-labs/solana/pull/18648/commits/b6803479a84fc66b5605396c5f52ad5f9400ec62